### PR TITLE
Sigoptlite test best assignments

### PR DESCRIPTION
**Pull request by malkomes**
- move best assignments tests to endpoint. We still need to work on correct integration tests but that will do for now
- couple of fixes

### CODE REVIEW

File: sigoptlite/sigoptlite/models.py [hunk 1]
```diff
@@ -231,11 +231,18 @@ class LocalObservation:
...
+        failed=False,
+      )
     return dict(
       assignments=dict(self.assignments),
-      values=[dataclass_to_dict(me) for me in self.get_metric_evaluations(experiment)],
+      values=None,
       task=dataclass_to_dict(self.task) if self.task else None,
-      failed=self.failed,
+      failed=True,
     )
```
Comment (harvey-intel) [inline sigoptlite/sigoptlite/models.py:246]: prefer to revert the if statement.

Could also just inline everything here?
```python
return dict(
  assignments=dict(self.assignments),
  values=None if self.failed else [dataclass_to_dict(me) for me in self.get_metric_evaluations(experiment)],
  task=dataclass_to_dict(self.task) if self.task else None,
  failed=self.failed,
)
```

**Review by harvey-intel — COMMENTED** [review]
Some nits. Need to remember to move the assignments to `connection` later.

File: sigoptlite/sigoptlite/models.py (hunk 1 above)
Comment (malkomes) [reply sigoptlite/sigoptlite/models.py:246]: yeah , good call!

**Review by erichanslee — APPROVED** [review]
Thanks for catching the error and moving over the test! 

I'm adding that we need to clean up the best assignments test in jira

**Pull request merged**